pyuvdata/uvdata.py

```diff
@@ -3726,6 +3778,38 @@ def read(self, filename, axis=None, file_type=None, allow_rephase=True,
                         run_check_acceptability=run_check_acceptability,
                         keep_all_metadata=keep_all_metadata)
 
+            elif file_type == 'mwa_corr_fits':
+                if (antenna_nums is not None or antenna_names is not None
+                        or ant_str is not None or bls is not None
+                        or frequencies is not None or freq_chans is not None
+                        or times is not None or polarizations is not None
+                        or blt_inds is not None):
+                    select = True
+                    warnings.warn(
+                        'Warning: select on read keyword set, but '
+                        'file_type is "mwa_corr_fits" which does not support select '
+                        'on read. Entire file will be read and then select '
+                        'will be performed')
+                else:
+                    select = False
+
+                self.read_mwa_corr_fits(
+                    filename, run_check=run_check,
+                    use_cotter_flags=use_cotter_flags,
+                    correct_cable_len=correct_cable_len,
+                    phase_data=phase_to_pointing_center,
+                    check_extra=check_extra,
+                    run_check_acceptability=run_check_acceptability)
```

Comment:
I think that either this should contain a `phase_center` keyword option, or we should remove that keyword from `read_mwa_corr_fits`. Maybe it makes sense to give the reader only a `phase_to_pointing_center` option. Or, maybe it makes sense to move phasing outside of the reader altogether.

Reply:
So the phase_center business is taken care of below, where the read method applies the requested phasing. We could remove it from the `read_mwa_corr_fits` altogether, but I think we need to keep the pointing center information because it is only known to that method. Let me think a little more about whether there's a clean way to get that information back to the read method so the phasing can be done in one place.

Reply:
I did what we discussed on slack -- I deprecated the `phase_center` keyword and changed (with deprecation warnings) the `phase_data` keyword to `phase_to_pointing_center`.

Reply:
Great, thanks!